modules/core/core-content/src/main/java/com/enonic/xp/core/impl/project/UpdateProjectRolesCommand.java

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    private void doSetRolesMembers( final Project oldProject, final Project newProject )
+    {
+        doSetRoleMembers( oldProject, newProject, ProjectPermissionsLevel.OWNER, ProjectRoles.OWNER );
+        doSetRoleMembers( oldProject, newProject, ProjectPermissionsLevel.EDITOR, ProjectRoles.EDITOR );
+        doSetRoleMembers( oldProject, newProject, ProjectPermissionsLevel.AUTHOR, ProjectRoles.AUTHOR );
+        doSetRoleMembers( oldProject, newProject, ProjectPermissionsLevel.CONTRIBUTOR, ProjectRoles.CONTRIBUTOR );
+    }
+
+    private void doSetRoleMembers( final Project oldProject, final Project newProject,
+                                   final ProjectPermissionsLevel projectPermissionsLevel, final ProjectRoles projectRole )
+    {
+
+        final Set<PrincipalKey> addedMembers;
+        final Set<PrincipalKey> removedMembers;
+
+        if ( oldProject == null )
+        {
+            addedMembers = newProject.getPermissions().getPermission( projectPermissionsLevel ).getSet();
+            removedMembers = Set.of();
+        }
+        else
+        {
+            addedMembers = doGetAddedMembers( newProject, projectPermissionsLevel, projectRole );
+            removedMembers = doGetRemovedMembers( oldProject.getPermissions(), newProject.getPermissions(), projectPermissionsLevel );
+        }
+
+        if ( !addedMembers.isEmpty() )
+        {
+            doAddRoleMembers( addedMembers, newProject.getName(), projectRole );
+        }
+        if ( !removedMembers.isEmpty() )
+        {
+            doRemoveRoleMembers( removedMembers, newProject.getName(), projectRole );
+        }
+    }
+
+    private void doAddRoleMembers( final Set<PrincipalKey> principalKeys, final ProjectName projectName, final ProjectRoles projectRole )
+    {
+        principalKeys.forEach( principalKey -> {
+            final PrincipalRelationship rel = PrincipalRelationship.
+                from( projectRole.getRoleKey( projectName ) ).
+                to( principalKey );
+
+            securityService.addRelationship( rel );
+        } );
+    }
+
+    private void doRemoveRoleMembers( final Set<PrincipalKey> principalKeys, final ProjectName projectName, final ProjectRoles projectRole )
+    {
+        principalKeys.
+            forEach( principalKey -> {
+
+                final PrincipalRelationship rel = PrincipalRelationship.
+                    from( projectRole.getRoleKey( projectName ) ).
+                    to( principalKey );
+
+                securityService.removeRelationship( rel );
+            } );
+    }
+
+    private Set<PrincipalKey> doGetAddedMembers( final Project newProject, final ProjectPermissionsLevel projectPermissionsLevel,
+                                                 final ProjectRoles projectRole )
+    {
+
+        final PrincipalKey roleKey = securityService.getRole( projectRole.getRoleKey( newProject.getName() ) ).
+            map( Role::getKey ).
+            orElse( null );
```

Comment:
replace orElse with ifPresent , so you don't need to deal with null checking at all.